### DISCUSSION TITLE
MIMO Full-Duplex

### DISCUSSION THREAD

**Discussion by Iligo** [NVlabs/sionna#137] [Q&A] [answered]
Hello,

I am writing to seek assistance with simulating Full-Duplex MIMO using Sionna. My main objective is to have a base station (BS) with multiple antennas transmitting and receiving data to/from different users simultaneously, within the same frequency. However, I am facing difficulties in implementing two specific features:

   1.- Self-Interference (SI): The messages sent by the BS to the users are leaked and interfere with the antennas receiving messages from those users.

   2.- Cross-Interference: Nearby users experience interference as some are receiving data while others are transmitting. All of this is happening using the same resources.

My ultimate goal is to propose beamforming at the BS to mitigate the self-interference. However, I am struggling to simulate this environment effectively. I have attempted to use "sionna.mimo.StreamManagement(rx_tx_association, num_streams_per_tx)" but have been unsuccessful in incorporating these interferences.

I would greatly appreciate any guidance on how to implement these interferences in Sionna, whether through built-in methods or any alternative approaches that may be more suitable for this problem. Thank you in advance for your assistance!

**Comment by jhoydis** [comment, marked as answer]
Hi,

Sionna does not have any specific layers for full-duplex systems. However, you can simply generate the channel impulse responses between all devices of interest and then use them to determine the received signals. These can be combined in any way you want to simulate interference.

That would mean in your case that you would need to generate:

1) Channels between the BS and all users
2) Channels between the users
3) Channels between the BS antennas

The stochastic channel models from 3GPP allow you to generate 1).  If you use ray tracing, you can generate 1) and 2). How you model 3) (and also 2)) is up to you and this might depend a lot on your assumptions.

The best way to model interference is by simply adding it to the received signals before receive processing. This is independent of the ``StreamManagement``.

In principle, you could setup a ``StreamManagement`` instance for the uplink and one for the downlink and model first interference free transmissions in both directions. Then, you would need to compute the interfering signals (by feeding the respective transmitted signals through the channels 2) and 3)) and add them to the received signals prior to decoding.

Hope that helps.

**Reply by Iligo** [reply]
Understood! Thanks for your fast and accurate response.
